File: CWNotificationDemo/MainViewController.m
//
//  MainViewController.m
//  CWNotificationDemo
//
//  Created by Cezary Wojcik on 11/15/13.
//  Copyright (c) 2013 Cezary Wojcik. All rights reserved.
//

#import "MainViewController.h"
#import "CWStatusBarNotification.h"

@interface MainViewController ()

@end

@implementation MainViewController

@synthesize lblDuration, txtNotificationMessage, sliderDuration;

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    self.title = @"CWStatusBarNotification";
    [self updateDurationLabel];
    UIFont *font = [UIFont boldSystemFontOfSize:10.0f];
    NSDictionary *attributes = [NSDictionary dictionaryWithObject:font forKey:NSFontAttributeName];
    [self.segFromStyle setTitleTextAttributes:attributes forState:UIControlStateNormal];
    [self.segToStyle setTitleTextAttributes:attributes forState:UIControlStateNormal];
    
    // initialize CWNotification
    self.notification = [CWStatusBarNotification new];
    
    // set default blue color (since iOS 7.1, default window tintColor is black)
    self.notification.notificationLabelBackgroundColor = [UIColor colorWithRed:0.0 green:122.0/255.0 blue:1.0 alpha:1.0];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)updateDurationLabel
{
    self.lblDuration.text = [NSString stringWithFormat:@"%f seconds", self.sliderDuration.value];
}

- (IBAction)sliderDurationChanged:(UISlider *)sender
{
    [self updateDurationLabel];
}

# pragma mark - show notification

- (void)setupNotification
{
    self.notification.notificationAnimationInStyle = self.segFromStyle.selectedSegmentIndex;
    self.notification.notificationAnimationOutStyle = self.segToStyle.selectedSegmentIndex;
    self.notification.notificationStyle = self.notificationStyle.selectedSegmentIndex == 0 ?
    CWNotificationStyleStatusBarNotification : CWNotificationStyleNavigationBarNotification;
}

- (IBAction)btnShowNotificationPressed:(UIButton *)sender
{
    [self setupNotification];
    [self.notification displayNotificationWithMessage:self.txtNotificationMessage.text forDuration:self.sliderDuration.value];
}

- (IBAction)btnShowCustomNotificationPressed:(UIButton *)sender {
    [self setupNotification];
    UIView *view = [[NSBundle mainBundle] loadNibNamed:@"CustomView" owner:nil options:nil][0];
    [self.notification displayNotificationWithView:view forDuration:self.sliderDuration.value];
}

@end

File: CWNotificationDemo/MainViewController.swift
//
//  MainViewController.swift
//  CWNotificationDemo
//
//  Created by Cezary Wojcik on 7/11/15.
//  Copyright © 2015 Cezary Wojcik. All rights reserved.
//

import UIKit
import CWStatusBarNotification

class MainViewController: UIViewController {
    // MARK: - IB outlets
    
    @IBOutlet weak var lblDuration: UILabel!
    @IBOutlet weak var sliderDuration: UISlider!
    @IBOutlet weak var txtNotificationMessage: UITextField!
    @IBOutlet weak var segFromStyle: UISegmentedControl!
    @IBOutlet weak var segToStyle: UISegmentedControl!
    @IBOutlet weak var segNotificationStyle: UISegmentedControl!
    
    // MARK: - properties
    
    let notification = CWStatusBarNotification()
    
    // MARK: - setup
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "CWStatusBarNotification"
        self.updateDurationLabel()
        
        // setup font
        let font = UIFont.boldSystemFont(ofSize: 10.0)
        let attributes = [NSFontAttributeName : font]
        self.segFromStyle.setTitleTextAttributes(attributes, for: UIControlState())
        self.segToStyle.setTitleTextAttributes(attributes, for: UIControlState())
        
        // set default blue color (since iOS 7.1, default window `tintColor`
        // is black)
        self.notification.backgroundColor = UIColor(red: 0.0,
            green: 122.0/255.0, blue: 1.0, alpha: 1.0)
    }
    
    // MARK: - methods
    
    func updateDurationLabel() {
        self.lblDuration.text = NSString(format: "%.1f seconds",
            self.sliderDuration.value) as String
    }
    
    func setupNotification() {
        guard let inStyle = CWNotificationAnimationStyle(rawValue:
            self.segFromStyle.selectedSegmentIndex) else {
                return
        }
        guard let outStyle = CWNotificationAnimationStyle(rawValue:
            self.segToStyle.selectedSegmentIndex) else {
                return
        }
        guard let notificationStyle = CWNotificationStyle(rawValue:
            self.segNotificationStyle.selectedSegmentIndex) else {
                return
        }
        self.notification.animationInStyle = inStyle
        self.notification.animationOutStyle = outStyle
        self.notification.style = notificationStyle
    }
    
    // MARK: - IB actions
    
    @IBAction func sliderDurationChanged(_ sender: UISlider) {
        self.updateDurationLabel()
    }
    
    @IBAction func btnShowNotificationPressed(_ sender: UIButton) {
        self.setupNotification()
        let duration = TimeInterval(self.sliderDuration.value)
        self.notification.displayNotification(message:
            self.txtNotificationMessage.text!, duration: duration)
    }
    
    @IBAction func btnShowCustomNotificationPressed(_ sender: UIButton) {
        self.setupNotification()
        let duration = TimeInterval(self.sliderDuration.value)
        guard let view = Bundle.main.loadNibNamed("CustomView",
            owner: nil, options: nil)?[0] as? UIView else {
                return
        }
        self.notification.displayNotification(view: view,
            forDuration: duration)
    }
    
}
